refnx/reflect/simulation.py

```diff
@@ -0,0 +1,167 @@
+        The path and name of the .pdb file
+    lgtfile: str
+        The path and name of the .lgt file (which contains the scattering
+        lengths of each of the atom type in the pdbfile). Currently the .lgt
+        file style that is supported is a 3 column space separated txt file
+        where the columns are atom_type, real_scattering_length, and
+        imaginary_scattering_length respectively.
+    xray: bool, optional
+          True if the scattering length of the particles should be scaled by
+          the classical radius of an electron.
+    layer_thickness: float
+        The thickness of the layers that the simulation cell should be sliced
+        into.
+    cut_off: float
+        The size of the simulation cell that should be ignored from the bottom.
+        This is to allow for the use of a vacuum gap at the bottom of the cell.
+    flip: bool, optional
+        False if the system should be read as is, true is the simulation cell
+        should be rotated through they-plane -- note that this treats the first
+        side that the neutron or X-ray interacts with as that at z=0.
+    roughness: float, optional
+        The fractional (of the layer thickness) roughness to be considered
+        between the layers.
+    verbose: bool, optional
+        True if you want to be notified when the pdb and lgt files have been
+        read and when the sld profile is calculated.
+    """
+    def __init__(self, pdbfile=None, lgtfile=None, xray=False,
+                 layer_thickness=1, cut_off=5, flip=False, roughness=0,
+                 verbose=False):
+        self.pdbfile = pdbfile
+        self.lgtfile = lgtfile
+        self.u = None
+        self.xray = xray
+        self.scatlens = {}
+        self.read_pdb()
+        if verbose:
+            print('PDB file read.')
+        self.read_lgt()
+        if verbose:
+            print('LGT file read.')
+        self.av_layers = np.zeros((int(self.u.dimensions[2] /
+                                       layer_thickness) -
+                                   cut_off, 5))
+        self.layers = np.zeros((self.av_layers.shape[0],
+                                self.av_layers.shape[1],
+                                len(self.u.trajectory)))
+        self.layers[:, 0, :] = layer_thickness
+        self.layers[:, 3, :] = layer_thickness * roughness
+        self.layers[:, 4, :] = 0
+        self.av_layers[:, 0] = layer_thickness
+        self.av_layers[:, 3] = layer_thickness * roughness
+        self.av_layers[:, 4] = 0
+        self.flip = flip
+        self.cut_off = cut_off
+        self.get_sld_profile()
+        if verbose:
+            print('SLD profile determined.')
+
+    def read_pdb(self):
+        """Parse pdb file.
+
+        This reads the pdb file into memory.
+        """
+        self.u = mda.Universe(self.pdbfile)
+
+    def read_lgt(self):
+        """Parses .lgt.
+
+        Parses the lgtfile. If no lgtfile is defined falass will help the user
```

Comment:
falass? :-)

Reply:
Haha, this is a residual from when I had my own code to do this. 
Resolved.